bolt-sidecar/src/api/commitments/firewall_recv.rs

```diff
@@ -0,0 +1,548 @@
+                        match message_res {
+                            Ok(Message::Text(_text)) => {
+                                // Create the channel to send and receive the commitment response
+                                let (tx, rx) = oneshot::channel();
+
+                                // TODO: parse the text into a commitment request
+                                let request =
+                                    CommitmentRequest::Inclusion(InclusionRequest::default());
+                                let event = CommitmentEvent { request, response: tx };
+
+                                if let Err(err) = this.api_events_tx.try_send(event) {
+                                    error!(?err, "failed to forward commitment event to channel");
+                                }
+
+                                // add the pending response to this buffer for later processing
+                                this.pending_commitment_responses.push(rx.boxed());
+                            }
+                            Ok(Message::Close(_)) => {
+                                warn!(?rpc_url, "websocket connection closed by server");
+                                return Poll::Ready(());
+                            }
+                            Err(e) => {
+                                error!(
+                                    ?e,
+                                    ?rpc_url,
+                                    "error reading message from websocket connection"
+                                );
+                                return Poll::Ready(());
+                            }
+                            _ => {} // Ignore non-text messages
+                        }
+                    }
+                    None => {
+                        warn!("websocket connection with {} closed by server", rpc_url);
+                        return Poll::Ready(());
+                    }
+                }
+            }
+
+            // 2. Handle commitment request responses after they've been processed.
+            while let Poll::Ready(Some(response)) =
+                this.pending_commitment_responses.poll_next_unpin(cx)
+            {
+                progress = true;
+                match response {
+                    Ok(commitment_result) => {
+                        if let Ok(commitment) = commitment_result {
+                            // TODO: check whether this format is correct + handle errors.
+                            let message =
+                                Message::text(serde_json::to_string(&commitment).unwrap());
+                            // Add the message to the outgoing messages queue
+                            this.outgoing_messages.push_back(message);
+                        }
+                    }
+                    Err(e) => {
+                        error!(?e, "failed to receive commitment response");
+                    }
```

Comment:
if this is a oneshot::error then we don't need to match on this, as it can only happen if the sender is dropped